_pytest/cacheprovider.py

```diff
@@ -147,11 +142,11 @@ def pytest_collection_modifyitems(self, session, config, items):
                     previously_failed.append(item)
                 else:
                     previously_passed.append(item)
-            if not previously_failed and previously_passed:
```

Comment:
if `test_a.py::test_a` is failed and you run `pytest test_b.py --lf` then you shouldn't remove passed tests from the test set, that way you can run in lf mode and work subsets of the failure set until they pass